### PR TITLE
Add support for regular expressions containing octal digits greater than `\200`

### LINKED ISSUE #4746: [FEA] Add support for regular expressions containing octal digits in range `\200` to `377`
**Describe the bug**
We now support octal digits but with limitations. We call back to CPU for certain ranges.

**Expected behavior**
We should support the full range of octal digits on the GPU.

### PR DESCRIPTION
**Pull request by anthony-chang**
Fixes #4746

cuDF only supports hexadecimal characters up to `\200`, and we fallback to CPU for characters outside that range. This workaround converts octal digits directly to the Unicode character itself, allowing GPU support for the full range of octal digits.

Signed-off-by: Anthony Chang <antchang@nvidia.com>

### CODE REVIEW

File: tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala [hunk 1]
```diff
@@ -184,6 +176,12 @@ class RegularExpressionTranspilerSuite extends FunSuite with Arm {
         "\u0007\u003f\u007f", "\u007f", "\u007f2"))
   }
 
+  test("octal digits >= 0o177 - find") {
```
Comment (NVnavkumar) [inline tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala:179]: I think should be merged this with other octal digit unit test to test all octal digits in one unit test.

**Review by NVnavkumar — COMMENTED** [review]
Looks good. Can you also add an integration test for this one?

**Comment by anthony-chang** [comment]
build

**Comment by anthony-chang** [comment]
build

**Review by NVnavkumar — APPROVED** [review]

**Pull request merged**